### PR TITLE
feat/ Socket functionality

### PR DESCRIPTION
**Pull request by APTy**
Reconfigured main server index.js to allow for sockets. Exposed the socket connection to a separate routing module to handle socket interaction. Wrote tests to confirm the above functionality.

### CODE REVIEW

**Comment by mrblueblue** [comment]
MERGE

File: specs/sockets.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  });
+
+  it('should authenticate the connection', function(done) {
+    connection.emit('init', testUser);
+    connection.once('score', function() {
+      done();
+    });
+  });
+
+  it('should return a user\'s score after receiving an upvote via socket event', function(done) {
```
Comment (bhardin) [inline specs/sockets.js:56]: Just curious:

Why not write it as:

``` javascript
it("should return a user's score after receiving an upvote via socket event"
```

So you don't need the annoying backslash. :)

Comment (amZotti) [inline specs/sockets.js:56]: Hey there,

Thanks for the feedback! The style guide our team decided on specifies we should use single quotes and we didn't want to be inconsistent.

Comment (bhardin) [inline specs/sockets.js:56]: lol. No problem. At least you know which way to argue for your styleguide, next time. :)

Hopefully see you next week.

**Pull request merged**